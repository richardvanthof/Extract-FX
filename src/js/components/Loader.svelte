<script>

</script>

<style>
    main {
        margin: auto;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        height: 75vh;
    }
</style>

<main>
    <progress></progress>
	<p class="caption">Loading...</p>
</main>