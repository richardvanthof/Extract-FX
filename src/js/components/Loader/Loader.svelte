<script lang='ts'>
    type Props = {
        message?: string,
        progress?: number
    }
    const {message = 'loading...', progress}: Props = $props();
</script>

<style>
    div {
        margin: auto;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        position: absolute;
        top: 0;
        left: 0;
        width: 100vw;
        height: 100vh;
        background: var(--bg-loader);
        box-shadow: 0 4px 30px rgba(0, 0, 0, 0.1); 
        backdrop-filter: blur(2px);
        -webkit-backdrop-filter: blur(5px);
    }
</style>

<div>
    <progress value={progress} ></progress>
	<p class="caption">{message}</p>
</div>