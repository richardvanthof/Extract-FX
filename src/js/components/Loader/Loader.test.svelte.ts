import { render } from '@testing-library/svelte';
import { screen } from '@testing-library/svelte';
import '@testing-library/jest-dom';
import {it, describe, expect } from 'vitest';
import Loader from './Loader.svelte';

describe('Loader Component', () => {
    type Props = {
        message?: string,
        progress?: number
    }

    it('Renders', ()=> {
        
        const {container} = render(Loader)

        const progress = screen.getByRole('progressbar');

        expect(progress).toBeInTheDocument();
        expect(container).toMatchSnapshot();

    });

    it('Shows status message', () => {
        const props:Props = {
            message: 'Extracting...'
        };
        const {container} = render(Loader, {props})

        const messageText = screen.getByText(props.message);

        expect(messageText).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    })
});
