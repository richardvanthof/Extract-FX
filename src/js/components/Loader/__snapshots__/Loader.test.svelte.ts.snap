// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader Component > Renders 1`] = `
<div>
  <div
    class="svelte-85y53y"
  >
    <progress />
     
    <p
      class="caption"
    >
      loading...
    </p>
  </div>
  
</div>
`;

exports[`Loader Component > Shows status message 1`] = `
<div>
  <div
    class="svelte-85y53y"
  >
    <progress />
     
    <p
      class="caption"
    >
      Extracting...
    </p>
  </div>
  
</div>
`;
